tests/functional/modules/test_zos_zfs_resize_func.py

```diff
@@ -0,0 +1,551 @@
+            cmd = "cat \"//'{0}'\" ".format(trace_destination_ds_s)
+            output_of_trace_file = hosts.all.shell(cmd=cmd)
+            for out in output_of_trace_file.contacted.values():
+                assert out.get("stdout") is not None
+
+    finally:
+        clean_up_environment(hosts=hosts, ds_name=ds_name, temp_dir_name=mount_folder)
+        if trace_destination == "pds" or trace_destination == "pdse":
+            trace_destination_ds = trace_destination_ds.split("(")[0]
+            trace_destination_ds_s = trace_destination_ds_s.split("(")[0]
+        hosts.all.zos_data_set(name=trace_destination_ds, state="absent")
+        hosts.all.zos_data_set(name=trace_destination_ds_s, state="absent")
+
+#########################
+# Negative test cases
+#########################
+
+@pytest.mark.parametrize("space_type", ["K", "TRK", "CYL", "M", "G"])
+def test_space_type_not_accept(ansible_zos_module, space_type):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 2500
+    results = hosts.all.zos_zfs_resize(target=ds_name,
+                                        size=size,
+                                        space_type=space_type)
+    for result in results.contacted.values():
+        assert result.get('failed') == True
+        assert result.get('msg') == "value of space_type must be one of: k, m, g, cyl, trk, got: {0}".format(space_type)
+
+def test_target_do_not_exist(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 2500
+    results = hosts.all.zos_zfs_resize(target=ds_name,
+                                        size=size,)
+    for result in results.contacted.values():
+        assert result.get('target') == ds_name
+        assert result.get('size') == size
+        assert result.get('rc') == 1
+        assert result.get('changed') is False
+        assert result.get('msg') == "ZFS Target {0} does not exist".format(ds_name)
+
+def test_mount_point_does_not_exist(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 2500
+    try:
+        hosts.all.zos_data_set(name=ds_name, type="zfs", space_primary=1, space_type="m")
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=size,)
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('size') == size
+            assert result.get('rc') == 1
+            assert result.get('changed') is False
+            assert  "No mount points were found in the following output:" in result.get('msg')
+    finally:
+        hosts.all.zos_data_set(name=ds_name, state="absent")
+
+def test_no_operation_executed(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    size = 1440
```

Comment:
aha! this seems to answer my above question about why 1400 was downsizing ? is zos_data_set creating datasets with 1440kb when specified 1mb?

Reply:
Yes, is no accurate, but is our system